src/main/java/net/dv8tion/jda/api/utils/concurrent/Task.java

```diff
@@ -39,6 +39,9 @@
      * Provide a callback for exception handling.
      * <br>This is an asynchronous operation.
      *
+     * <p>The error will be logged regardless of your callback, this only exists to handle
```

Comment:
This isn't technically guaranteed because it depends on whether the Future has had an `.exceptionally` registered to it _before_ it gets passed to the Task.

Reply:
We *are* doing that though. The exception handler is already active before we create the `GatewayTask` instance. If some other implementation of the interface doesn't do that, then it violates the contract of this method.